src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -3392,5 +3392,29 @@ private RestoreCommandProvidersCache GetRestoreProviderCache()
 
             return _restoreProviderCache;
         }
+
+        public static async Task<HashSet<string>> GetTargetFramework(NuGetProject project)
+        {
+            var frameworks = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            NuGetFramework framework;
+            if (project.TryGetMetadata<NuGetFramework>(NuGetProjectMetadataKeys.TargetFramework, out framework))
+            {
+                frameworks.Add(framework.ToString());
+            }
+
+            if (project is BuildIntegratedNuGetProject)
```

Comment:
You can use `as` to check and assign to a new variable.

Reply:
Fixed